### PR TITLE
MSI-2599: Clearing index with wrong predicate

### PR DESCRIPTION
**Pull request by Stepa4man**
InventoryConfigurableProductIndexer uses all columns value in predicate for clearing index

### Description
Described in Issue

### Fixed Issues
1. magento/inventory#2599: InventoryConfigurableProductIndexer uses all columns value in predicate for clearing index

### Manual testing scenarios
1. Described in Issue

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/InventoryIndexer/Indexer/Stock/PrepareIndexDataForClearingIndex.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+     *
+     * @param ArrayIterator $indexData
+     *
+     * @return ArrayIterator
+     */
+    public function execute(ArrayIterator $indexData): ArrayIterator
+    {
+        $clearIndex = [];
+        foreach ($indexData as $indexItem) {
+            $clearIndex[]['sku'] = $indexItem['sku'];
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryIndexer/Indexer/Stock/PrepareIndexDataForClearingIndex.php:27]: Can we use [`array_column`](https://www.php.net/manual/en/function.array-column.php)?

Comment (Stepa4man) [reply app/code/Magento/InventoryIndexer/Indexer/Stock/PrepareIndexDataForClearingIndex.php:27]: No. Returning structure will be different.

**Review by ishakhsuvarov — APPROVED** [review]

**Comment by elvinristi** [comment]
FYI @Stepa4man 

PR description's `Fixed Issues` section has incorrect project fix reference / should be `magento/inventory#2599` and not `magento/magento#2599`

**Comment by Stepa4man** [comment]
> FYI @Stepa4man
> 
> PR description's `Fixed Issues` section has incorrect project fix reference / should be `magento/inventory#2599` and not `magento/magento#2599`

Hey @elvinristi Thanks thanks for heads up. Fixed

**Pull request merged**